src/main/java/org/jenkinsci/plugins/configfiles/maven/security/ServerCredentialMapping.java

```diff
@@ -45,19 +52,27 @@ public String getCredentialsId() {
     @Extension
     public static class DescriptorImpl extends Descriptor<ServerCredentialMapping> {
 
-        public ListBoxModel doFillCredentialsIdItems(@AncestorInPath AbstractProject<?, ?> context, @QueryParameter String serverId) {
-            if ( (context == null && Jenkins.getInstance().hasPermission(Jenkins.ADMINISTER))
-                    || (context != null && context.hasPermission(Item.CONFIGURE)) ) {
-                final List<StandardUsernameCredentials> validCredentials = CredentialsHelper.findValidCredentials(serverId);
-                return new StandardUsernameListBoxModel().withEmptySelection().withAll(validCredentials);
+        public ListBoxModel doFillCredentialsIdItems(@AncestorInPath ItemGroup context, @QueryParameter String serverId) {
+            AccessControlled _context = (context instanceof AccessControlled ? (AccessControlled) context : Jenkins.getInstance());
+            if (_context == null || !_context.hasPermission(Computer.CONFIGURE)) {
+                return new StandardUsernameListBoxModel().includeCurrentValue(serverId);
             }
-            return new ListBoxModel();
+
+            // @formatter:off
+            return new StandardUsernameListBoxModel().includeAs(
+                        ACL.SYSTEM, 
```

Comment:
`context instanceof Queue.Task ? Tasks.getDefaultAuthenticationOf((Queue.Task)context) : ACL.SYSTEM` is probably what you want rather than a raw `ACL.SYSTEM`